### PR TITLE
feat(workflows):product handlers should be reusable in different context

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
The listProducts handler should not be specific to the workflow.
Same for the product removal, shouldn't expect an entire DTO but just a collection of object that at least contains the id.
Same principle applied to other product handlers

### CODE REVIEW

File: packages/workflows/src/definition/create-products.ts [hunk 1]
```diff
@@ -275,10 +275,16 @@ const handlers = new Map([
             },
             {
               from: Actions.createProducts,
-              alias: ProductHandlers.listProducts.aliases.products,
+              alias: "products",
```
Comment (riqwan) [inline packages/workflows/src/definition/create-products.ts:278]: q: whats the reasoning behind removing the alias?

File: packages/workflows/src/handlers/product/attach-sales-channel-to-products.ts [hunk 2]
```diff
@@ -10,7 +9,7 @@ export async function attachSalesChannelToProducts({
   data,
 }: WorkflowArguments<{
   productsHandleSalesChannelsMap: Map<ProductHandle, SalesChannelId[]>
-  products: ProductTypes.ProductDTO[]
+  products: { handle: string; id: string }[]
```
Comment (riqwan) [inline packages/workflows/src/handlers/product/attach-sales-channel-to-products.ts]: yaas, i like this. I ended up wanting to do the same thing. 

suggestion: Maybe put this as a type at the beginning of the file?

like: `WorkflowArguments<HandlerOutputData>`

**Review by riqwan — APPROVED** [review]
lgtm! 🔥

File: packages/workflows/src/definition/create-products.ts (hunk 1 above)
Comment (riqwan) [reply packages/workflows/src/definition/create-products.ts:278]: I see you removed the handler.

Comment (adrien2p) [reply packages/workflows/src/definition/create-products.ts:278]: exactly, I only need it here for the middleware

File: packages/workflows/src/handlers/product/attach-sales-channel-to-products.ts (hunk 2 above)
Comment (adrien2p) [reply packages/workflows/src/handlers/product/attach-sales-channel-to-products.ts]: I ve extracted a portion of it, but yeah I can do that 👍

**Review by olivermrbl — COMMENTED** [review]
Should we move `type PartialProduct = { handle: ProductHandle; id: string }` to a type-file for reusability?

File: packages/workflows/src/handlers/product/list-products.ts [hunk 3]
```diff
@@ -1,17 +1,20 @@
 import { ProductTypes, WorkflowTypes } from "@medusajs/types"
 import { WorkflowArguments } from "../../helper"
 
+type HandlerInput = {
+  productIds: string[]
```
Comment (olivermrbl) [inline packages/workflows/src/handlers/product/list-products.ts]: suggestion: `ids` should be fine, the handler name speaks for itself
```suggestion
  ids: string[]
```

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Comment by adrien2p** [comment]
> Should we move `type PartialProduct = { handle: ProductHandle; id: string }` to a type-file for reusability?

I didn't because it is the handler typing, so if the handler needs to evolve and we are having a shared type, we will quickly end up composing custom stuff on top of it. I prefer if it stays to the handler as is for now. wdyt?

**Comment by olivermrbl** [comment]
> I prefer if it stays to the handler as is for now. wdyt?

Makes sense. Let's keep it as is 👍

**Pull request merged**